libsolutil/CommonData.cpp

```diff
@@ -188,5 +188,39 @@ string solidity::util::formatAsStringOrNumber(string const& _value)
 		if (c <= 0x1f || c >= 0x7f || c == '"')
 			return "0x" + h256(_value, h256::AlignLeft).hex();
 
-	return "\"" + _value + "\"";
+	return escapeAndQuoteString(_value);
```

Comment:
Call me paranoid, but I think we should have the scanner/parser rescan our results here and assert that the values are the same here. Not sure how easy this is, but still.

Reply:
Sounds good!

Reply:
``Scanner(CharStream("\"" + _value + "\"", "")).currentLiteral()`` or something like that? But yeah, maybe fine without.